x509/src/certificate.rs

```diff
@@ -80,25 +139,37 @@ pub struct TBSCertificate<'a> {
 impl<'a> ::der::Decodable<'a> for TBSCertificate<'a> {
     fn decode(decoder: &mut ::der::Decoder<'a>) -> ::der::Result<Self> {
         decoder.sequence(|decoder| {
-            let version =
+            let version_int =
                 ::der::asn1::ContextSpecific::decode_explicit(decoder, ::der::TagNumber::N0)?
                     .map(|cs| cs.value)
                     .unwrap_or_else(default_zero_u8);
+            let version = Version::try_from(version_int)?;
```

Comment:
If you change `Version` to impl `DecodeValue` you should be able to do:

```suggestion
            let version = decoder.context_specific::<Version>(der::TagNumber::N0, der::TagMode::Explicit)?;
```

Reply:
That won't work straight up because it yields an Option<> (and the version guards on the v2/v3 bits were on the int).

Reply:
I tried that unsuccessfully, will pick this back up tomorrow.

Reply:
Addressed this as well as default version handling in general in light of change from u8 to version: https://github.com/carl-wallace/formats/commit/b600b9982e63c0a39a13edd8bf0f29bd0969ba83